phpunit.xml.dist

```diff
@@ -4,17 +4,15 @@
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
          backupGlobals="false"
+         failOnDeprecation="true"
```

Comment:
if you fail on deprecations, you should also enable the `displayDeprecations` flag. Otherwise, the failure will be impossible to understand.

Reply:
based on our existing usage (and the behavior of older versions of PHPUnit), we should probably also fail on warnings and notices. Our debug mode generally breaks for them, so we want the testsuite to fail for them as well.

Reply:
> if you fail on deprecations, you should also enable the `displayDeprecations` flag. Otherwise, the failure will be impossible to understand.

From a local test, I think this is not needed. `failOn*` automatically enables `display*` (at least, I get not more output with the config value). The `display*` options seem to be useful when you don't want to fail on them, but still see the information.

Reply:
oh, I missed that https://github.com/sebastianbergmann/phpunit/issues/5856 has been fixed.